### ISSUE TITLE
SearchCompiler Error

### ISSUE THREAD

**Issue by Rub21** [Gubaer/josm-scripting-plugin#32]
I am trying to run : 
```js
var layer = josm.layers.activeLayer.data;
var result =dataset.query("type:relation");
```

Does anyone knows why? , I am using 8942version of JOSM

![image](https://cloud.githubusercontent.com/assets/1152236/10713036/1296544c-7a73-11e5-81cf-8f7de556000b.png)

cc. @Gubaer @alex85k

**Comment by Gubaer** [comment]
Probably the method `compile` has been renamed or moved in the JOSM code base. I'll have a look.

**Comment by don-vip** [comment]
Yep, see https://josm.openstreetmap.de/changeset/8811/josm/trunk/src/org/openstreetmap/josm/actions/search/SearchCompiler.java

**Comment by Gubaer** [comment]
Fixed in 7de6961. Published new plugin release 30722. Just update the plugin in the JOSM preferences dialog, provided you have JOSM >= 8906.

Try this script:
```javascript
var console = require("josm/scriptingconsole");
var data = josm.layers.activeLayer.data;
var result = data.query("type:relation");

console.println("num relations found: " + result.length);
```
